added ajax refresh button automatic integration with every GridField instance

jelicanin opened the pull request:
This change make ajax refresh button available on every GridField instance by default configuration.

[NOTE] in order to use this feature you must apply patch: 
API Change: Make GridFieldConfig objects decoratable #2311 
https://github.com/silverstripe/silverstripe-framework/pull/2311

jelicanin committed:
updated button positioning

jelicanin committed:
added ajax refresh button automatic integration with every GridField instance

[NOTE] in order to use this feature you must apply patch: https://github.com/silverstripe/silverstripe-framework/pull/2311

jelicanin committed:
Merge branch 'master' of https://github.com/silverstripe-labs/silverstripe-gridfieldajaxrefresh

# By Stephen Wake (2) and Steve Wake (1)
# Via Stephen Wake (2) and Julian Seidenberg (1)
* 'master' of https://github.com/silverstripe-labs/silverstripe-gridfieldajaxrefresh:
  Get the grid field id from the child element so works if more than one gridfield on the page
  Get the grid field id from the child element so works if more than one gridfield on the page
  Updated composer.json

jelicanin commented:
this patch is applicable now.
GridFieldConfig has been promoted to first-class Object

tractorcow commented on code/GridFieldAjaxRefreshConfig.php:
```diff
@@ -0,0 +1,22 @@
+class GridFieldAjaxRefresh_Base extends Extension {
+
+	public function updateConfig() {
+		$this->owner->addComponent(new GridFieldAjaxRefresh(5000, false));
```
Should we make this value a configurable option? (`private static`).

candidasa replied:
@tractorcow certainly would make sense to make this a config option. In the yml config system

dhensby commented:
@jelicanin it'd be great to get this over the line, can you make the setting a config option?

thanks

tractorcow commented:
Ping me if this doesn't happen, and maybe I'll find time to pull it up myself.

jelicanin commented:
sorry for late reply guys :(
I will fix it!

dhensby commented:
thanks @jelicanin

jelicanin committed:
Merge branch 'master' of https://github.com/silverstripe-labs/silverstripe-gridfieldajaxrefresh

jelicanin committed:
Added config params for
- auto refresh interval
- auto refresh enable / disable

GridFieldAjaxRefreshConfig Class Converted to PSR-2

jelicanin commented:
Added config params for
- auto refresh interval
- auto refresh enable / disable

jelicanin commented:
ping :)
